crypto/x509/x509_vfy.c

```diff
@@ -590,8 +594,12 @@ static int check_extensions(X509_STORE_CTX *ctx)
             }
         }
 
+        if (eku != 0 && i < ctx->num_untrusted)
```

Comment:
```suggestion
        if (eku != NID_undef)
```
Check the whole chain, not just the untrusted certificates.